src/routes/auth.ts

```diff
@@ -0,0 +1,29 @@
+import express, { Router } from 'express';
+import passport from 'passport';
+import { OAuth2Strategy } from 'passport-oauth';
+import { UserDAO } from '../data/userDAO';
+
+const {
+    DISCORD_CLIENT_ID,
+    DISCORD_CLIENT_SECRET
+} = process.env;
+
+export function getRouter(users: UserDAO): Router {
+    const router = express.Router();
+
+    passport.use('discord', new OAuth2Strategy({
+        authorizationURL: 'https://discordapp.com/api/oauth2/authorize',
+        tokenURL: 'https://discordapp.com/api/oauth2/token',
+        clientID: DISCORD_CLIENT_ID,
+        clientSecret: DISCORD_CLIENT_SECRET,
+        callbackURL: 'http://localhost:3000/auth/discord/callback'
```

Comment:
I need to update this callbackURL to not use localhost when it's in production.

maybe we need dev and prod environments set up?

Reply:
Putting it in the .env file would do the trick

Reply:
True yeah 🤔 that's better than nothing. I want to put it behind my own domain instead of the default heroku one too